java/ql/lib/semmle/code/java/security/MaybeBrokenCryptoAlgorithmQuery.qll

```diff
@@ -34,11 +37,44 @@ private predicate objectToString(MethodCall ma) {
   )
 }
 
+private class GetPropertyMethodCall extends MethodCall {
+  GetPropertyMethodCall() { this.getMethod() instanceof PropertiesGetPropertyMethod }
+
+  private ConfigPair getPair() {
+    this.getArgument(0).(ConstantStringExpr).getStringValue() = result.getNameElement().getName()
+  }
+
+  string getPropertyValue() {
+    result = this.getPair().getValueElement().getValue() or
+    result = this.getArgument(1).(ConstantStringExpr).getStringValue()
+  }
+}
+
+/**
+ * Get the string value represented by the given expression.
+ *
+ * If the value is a string literal, return the literal value.
+ * If the value is a call to `java.util.Properties::getProperty`, return the potential values of the property.
+ */
+string insecureAlgorithmName(DataFlow::Node algo) {
```

Comment:
Sorry, just a quick thing before approving:

This predicate should be named differently, since it does nothing related to `insecure`, it just gets the string value of the node `algo` if it's either a `StringLiteral` or a `GetPropertyMethodCall`. So I'd suggest something like:

```suggestion
/**
 * Gets the string value represented by the given expression node.
 *
 * If the value is a string literal, gets the literal value.
 * If the value is a call to `java.util.Properties::getProperty`, gets the potential values of the property.
 */
string getStringValue(DataFlow::ExprNode algo) {
```

Reply:
(Principle being name things for what they do, not for what they'll be used for)